sdk/identity/src/token_credentials/imds_managed_identity_credentials.rs

```diff
@@ -18,7 +18,27 @@ const MSI_API_VERSION: &str = "2019-08-01";
 /// This authentication type works in Azure VMs, App Service and Azure Functions applications, as well as the Azure Cloud Shell
 ///
 /// Built up from docs at [https://docs.microsoft.com/azure/app-service/overview-managed-identity#using-the-rest-protocol](https://docs.microsoft.com/azure/app-service/overview-managed-identity#using-the-rest-protocol)
-pub struct ImdsManagedIdentityCredential;
+#[derive(Default)]
+pub struct ImdsManagedIdentityCredential {
+    object_id: Option<String>,
+    client_id: Option<String>,
+    msi_res_id: Option<String>,
+}
+
+impl ImdsManagedIdentityCredential {
+    /// Create a new ImdsManagedIdentityCredential with the given optional parameters. Only one of object_id, client_id and msi_res_id may be set.
+    pub fn new(
```

Comment:
I don't think this constructor is super helpful. What do you think about removing it and adding [setters like Java SDK](https://github.com/Azure/autorest-clientruntime-for-java/blob/f5e7994a7eb914465525734021f6abfef7e6cd07/azure-client-authentication/src/main/java/com/microsoft/azure/credentials/MSICredentials.java#L92-L127). They would be like `with_object_id(object_id: String)`, with_client_id, and with_msi_red_id.

Reply:
That makes a lot of sense, I have made the change and got rid of the new error I had added, as setting the other id parameters to None eliminates the chance of having more than one set.